chore(broccoli): improve `overwrite` errors in merge-trees

caitp opened the pull request:
Report the relative path causing the error

caitp commented:
@btford was the relative path enough to figure out the offending path? If so a quick lgtm would be good

IgorMinar commented on tools/broccoli/broccoli-merge-trees.ts:
```diff
@@ -16,6 +16,11 @@ function outputFileSync(sourcePath, destPath) {
 
+function pathOverwrittenError(path) {
+  return new Error(
+      `duplicate path '${path}'.\n\`overwrite\` option is required for handling duplicates.`);
```
how about something more descriptive like:

`Duplicate path found while merging trees. Path: '....'. Either remove the duplicate or enable the 'overwrite' option for this merge.`

caitp replied:
done

caitp committed:
chore(broccoli): improve `overwrite` error in merge-trees

Modified the error message to include the relative duplicate path,
to help in diagnosing the cause of the error message.
